src/Uno.UI/UI/Xaml/FrameworkElement.macOS.cs

```diff
@@ -33,15 +33,38 @@ public override bool NeedsLayout
 					base.NeedsLayout = value;
 				}
 
-				RequiresMeasure = true;
-				RequiresArrange = true;
-
 				if (ShouldInterceptInvalidate)
 				{
 					return;
 				}
+			}
+		}
+
+		protected internal override void OnInvalidateMeasure()
```

Comment:
Does this logic need to be in the `IFrameworkElementHelper` method? Otherwise it seems like `IFrameworkElements` won't invalidate their parent correctly.

Reply:
Yes, it'll most likely fail. Will be addressed in a later PR as well.